veracruz-client/src/veracruz_client.rs

```diff
@@ -281,12 +257,10 @@ impl VeracruzClient {
         }
     }
 
-    pub fn get_results(&mut self) -> Result<Vec<u8>, VeracruzClientError> {
-        self.check_role_permission(&VeracruzRole::ResultReader)?;
+    pub fn get_results(&mut self, file_name:&str) -> Result<Vec<u8>, VeracruzClientError> {
```

Comment:
To be honest, I'm finding `get_results` to be quite unintuitive now. I originally expected `file_name` to be the name of the output file, but it's currently the executable.

Would it make more sense to split this into separate `execute(executable_name)` and `get_result(output_name)` functions?

Reply:
That is a good point. I am still thinking of it at this point. We might allow the runtime automatically execute program, if the input of this program has changed, for example. This function basically matches the structure of protocol buffer.

Reply:
I was thinking split this all the way down to `execution_engine_manager`, where "output" is hard-coded.

One option is you could leave it up to the user to decide who calls `execute`, I imagine in most cases it would be the same device that calls `get_result`. `get_result` could error with file-not-found if `execute` hasn't been called. This would match existing filesystem semantics the closest.

If you go for automatic execution, you could make a sort of dependency system, where you either execute eagerly (when all inputs are available) or lazily (when output is requested). Though I don't know if that would map exactly onto the rwx permissions, for example if you had two executables that wrote to the same file (a log?).

Re-executing when inputs change could also work, though it does raise some questions about if it's ok to use those resources if the executable is long-running/expensive.

Reply:
The split makes sense to me. If you feel bored of working on cortex. You are more than welcome to open another merge request after mine.

The program capability table in the policy can be served as a dependency table for now as a first draft. Obviously it is an extreme simple one. However I do not know how to handle streaming data. We could execute the program when all streaming data are updated. Note that in the streaming cases, client might append file instead overwrite file. 

Re-executing can work if a program is exposed with wasi API as the program should have full control of the resource, for example, which byte the program wants to read and the runtime just blocking and wait if necessary. Yet, there are a lot of details I do not have a proper think.